template.yaml

```diff
@@ -1,67 +1,87 @@
     Author: "Sage-Bionetworks"
     SpdxLicenseId: "Apache-2.0"
     # paths are relative to .aws-sam/build directory
     LicenseUrl: "LICENSE"
     ReadmeUrl: "README.md"
     Labels: ["serverless", "template", "github", "quick-start"]
-    HomePageUrl: "https://github.com/Sage-Bionetworks-IT/lambda-template"
-    SemanticVersion: "0.0.3"
-    SourceCodeUrl: "https://github.com/Sage-Bionetworks-IT/lambda-template/tree/0.0.3"
-
-# More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
-Globals:
-  Function:
-    Timeout: 3
+    HomePageUrl: "https://github.com/Sage-Bionetworks-IT/cfn-cr-same-region-bucket-download"
+    SemanticVersion: "0.0.1"
+    SourceCodeUrl: "https://github.com/Sage-Bionetworks-IT/cfn-cr-same-region-bucket-download/tree/0.0.1"
 
 Resources:
-  HelloWorldFunction:
-    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
-    Properties:
-      CodeUri: .
-      Handler: hello_world/app.lambda_handler
-      Runtime: python3.7
-      Role: !GetAtt FunctionRole.Arn
-      Events:
-        HelloWorld:
-          Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
-          Properties:
-            Path: /hello
-            Method: get
-
-  FunctionRole:   # execute lambda function with this role
-    Type: AWS::IAM::Role
+  RestrictBucketDownloadRegionRole:
+    Type: "AWS::IAM::Role"
     Properties:
       AssumeRolePolicyDocument:
-        Version: 2012-10-17
+        Version: "2012-10-17"
         Statement:
           - Effect: Allow
             Principal:
-              Service:
-                - lambda.amazonaws.com
-            Action:
-              - sts:AssumeRole
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Path: "/"
       ManagedPolicyArns:
         - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      Policies:
+        - PolicyName: external-bucket-lambda-policy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "s3:GetBucketPolicy"
+                  - "s3:PutBucketPolicy"
+                  - "s3:DeleteBucketPolicy"
+                Resource: "arn:aws:s3:::*"
+  RestrictBucketDownloadRegionFunction:
+    Type: "AWS::Serverless::Function"
+    Properties:
+      Description: Dynamically updates the external bucket policy to restrict downloading only IP addresses listed by AWS.
+      Runtime: python3.7
+      Handler: "restrict_download_region/restrict_region.handler"
+      Role: !GetAtt RestrictBucketDownloadRegionRole.Arn
+      Timeout: 30
+      Environment:
+        Variables:
+          REGION: !Ref "AWS::Region"
```

Comment:
you don't need to pass in the region, it's already set as a `Reserved environment variables`, https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html

Reply:
Already removed in the latest commit
https://github.com/zimingd/cfn-cr-same-region-bucket-download/blob/5cd47ec1d6b2c640c1d7df61e2cf9a63b693a8a5/template.yaml#L74